delfin/drivers/hitachi/hnas/nas_handler.py

```diff
@@ -0,0 +1,825 @@
+                    model = vendor = version = None
+                    if len(type_array) > 2:
+                        model = type_array[1].replace('Model', '')
+                        vendor = type_array[0].replace('Make', '')
+                        version = type_array[2].replace('Revision', '')
+                    pool_id = disk_map.get('Usedinspan', None)
+                    serial_number = disk_map['Luid'].split(']')[-1]
+                    if pool_id:
+                        pool_id = pool_id.split('(')[0]
+                    disk_model = {
+                        'name': disk_map['HDSdevname'],
+                        'storage_id': storage_id,
+                        'native_disk_id': disk_map['DeviceID'],
+                        'serial_number': serial_number,
+                        'manufacturer': vendor,
+                        'model': model,
+                        'firmware': version,
+                        'speed': None,
+                        'capacity': int(self.get_size(size)),
+                        'status': status,
+                        'physical_type': None,
+                        'logical_type': None,
+                        'native_disk_group_id': pool_id,
+                        'location': disk_map['Serialnumber'],
+                    }
+                    disks_list.append(disk_model)
+            return disks_list
+        except exception.DelfinException as e:
+            err_msg = "Failed to get disk from " \
+                      "hitachi nas: %s" % (six.text_type(e.msg))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get disk from " \
+                      "hitachi nas: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def get_pool_size(self):
+        size_info = self.ssh_do_exec([constant.POOL_SIZE_COMMAND])
+        size_array = size_info.split("\r\n")
+        size_map = {}
+        pool_name = None
+        count = 0
+        for size in size_array:
+            if 'Span ' in size:
+                pool_name = size.split()[-1].replace(':', '')
+                size_map[pool_name] = 0
+                count = 0
+            if '[Free space]' in size:
+                free_array = size.split()
+                if len(free_array) > 2:
+                    count += 1
+                    free_size = free_array[0].replace('GiB', 'GB')
+                    size_map[pool_name] += self.get_size(free_size)
+        return size_map
+
+    def get_pool(self, storage_id):
+        try:
+            pool_info = self.ssh_do_exec([constant.POOL_INFO_COMMAND])
+            pool_list = []
+            pool_array = self.get_table_data(pool_info)
+            size_map = self.get_pool_size()
+            for pool in pool_array:
+                value_array = pool.split()
+                if len(value_array) == 6:
```

Comment:
What's the `6` mean? Why only the length is `6` then we can add it to pool?

Reply:
Some useless information will be obtained through SSH.the information in each storage pool is larger than 6 columns, the useless information that is smaller than 6 columns is excluded to avoid dirty data